src/services/GenerateVerifiableCredential.ts

```diff
@@ -0,0 +1,457 @@
+  			});
+  	}
+  }
+	
+
+	/**
+		If the ID_DOCUMENT_AUTHENTICITY object has a recommendation value of approve, the document should be given a validity score of 2
+		If the ID_DOCUMENT_AUTHENTICITY has an recommendation value of approve and the object also includes an NFC check, the validity score will be 3. 
+	*/
+  private calculateValidityScore(
+  	authenticityRecommendation: string,
+  	documentContainsValidChip: boolean,
+  ): number {
+  	if (authenticityRecommendation === YotiSessionDocument.APPROVE) {
+  		return documentContainsValidChip ? 3 : 2;
+  	}
+  	return 0;
+  }
+	
+	/**
+		If the ID_DOCUMENT_FACE_MATCH object has a recommendation value of APPROVE, the document should be given a verification score of 3 
+	*/
+  private calculateVerificationProcessLevel(faceMatchCheck: string): number {
+  	return faceMatchCheck === YotiSessionDocument.APPROVE ? 3 : 0;
+  }
+	
+  private getCounterIndicator(
+  	ID_DOCUMENT_FACE_MATCH_RECOMMENDATION: any,
+  	ID_DOCUMENT_AUTHENTICITY_RECOMMENDATION: any,
+  	IBV_VISUAL_REVIEW_CHECK_RECOMMENDATION?: any,
+  	DOCUMENT_SCHEME_VALIDITY_CHECK_RECOMMENDATION?: any,
+  	PROFILE_DOCUMENT_MATCH_RECOMMENDATION?: any,
+  ): any {
+  	const ci: string[] = [];
+	
+  	const addToCI = (code: string | string[]) => {
+  		if (Array.isArray(code)) {
+  			ci.push(...code);
+  		} else {
+  			ci.push(code);
+  		}
+  	};
+	
+  	const handleFaceMatchRejection = () => {
+  		const { value, reason } = ID_DOCUMENT_FACE_MATCH_RECOMMENDATION;
+  		if (value === "REJECT") {
+  			switch (reason) {
+  				case "FACE_NOT_GENUINE":
+  				case "LARGE_AGE_GAP":
+  				case "PHOTO_OF_MASK":
+  				case "PHOTO_OF_PHOTO":
+  				case "DIFFERENT_PERSON":
+  					addToCI("V01");
+  					break;
+  				default:
+  					break;
+  			}
+  		}
+  	};
+	
+  	const handleAuthenticityRejection = () => {
+  		const { value, reason } = ID_DOCUMENT_AUTHENTICITY_RECOMMENDATION;
+  		if (value === "REJECT") {
+  			switch (reason) {
+  				case "COUNTERFEIT":
+  					addToCI("D14");
+  					break;
+  				case "EXPIRED_DOCUMENT":
+  					addToCI("D16");
+  					break;
+  				case "FRAUD_LIST_MATCH":
+  					addToCI(["F03", "D14"]);
+  					break;
+  				case "DOC_NUMBER_INVALID":
+  					addToCI("D02");
+  					break;
+  				case "TAMPERED":
+  				case "DATA_MISMATCH":
+  				case "CHIP_DATA_INTEGRITY_FAILED":
+  				case "CHIP_SIGNATURE_VERIFICATION_FAILED":
+  				case "CHIP_CSCA_VERIFICATION_FAILED":
+  					addToCI("D14");
+  					break;
+  				default:
+  					break;
+  			}
+  		}
+  	};
+	
+  	handleFaceMatchRejection();
+  	handleAuthenticityRejection();
+	
+  	return ci;
+  }
+
+  private attachPersonName(
+  	credentialSubject: VerifiedCredentialSubject,
+  	givenName: string,
+  	familyName: string,
+  ): VerifiedCredentialSubject {
+  	const givenNames = givenName.split(" ");
+  	const nameParts = givenNames.map((name) => ({ value: name, type: "GivenName" }));
+	
+  	nameParts.push({ value: familyName, type: "FamilyName" });
```

Comment:
Please check with Anand on the requirement around this. DCMAW recently had to implement a change where surnames where also split in to an array of individual names where the surname contains spaces - I expect we should do this here too

Reply:
Having spoken to F2F-BA team, we will explore this requirement in a future ticket

Reply:
OK, on the basis that what we are merging now is the same behaviour as is currently in DCMAW